### PR TITLE
[docs] Add documentation for leader lease configuration

### PR DESCRIPTION
**Pull request by ddhodge**
Add --leader_lease_duration_ms flag
See #13788

@netlify /preview/reference/configuration/yb-master/#leader-lease-duration-ms

### CODE REVIEW

**Comment by tylarb** [comment]
@ddhodge   content looks :+1:  

Needs to additionally be in yb-tserver documentation though - 

We maybe should add a note to each of these leader election flags saying "the following flags need to be identical between master and tserver"

**Comment by tylarb** [comment]
@ddhodge   can you sync with someone from DST  team on what the list of "following flags should be the same" should be? All of the follower settings related to this thread... but it's worth a review if there's a couple more

**Comment by ddhodge** [comment]
Will be adding to TServer doc and backporting to previous versions once wording is finalized.

File: docs/content/preview/reference/configuration/yb-master.md [hunk 1]
```diff
@@ -287,6 +287,14 @@ For read replica clusters, set the value to `10` in all `yb-tserver` and `yb-mas
 
 Default: `6`
 
+##### --leader_lease_duration_ms
+
+The leader lease duration, in milliseconds. A leader keeps establishing a new lease or extending the existing one with every consensus update. A new server is not allowed to serve as a leader (that is, serve up-to-date read requests or acknowledge write requests) until a lease of this duration has definitely expired on the old leader's side.
```
Comment (robertsami) [inline docs/content/preview/reference/configuration/yb-master.md:292]: > A new server is not allowed to serve as a leader (that is, serve up-to-date read requests or acknowledge write requests) until a lease of this duration has definitely expired on the old leader's side.

this isn't completely accurate, as we have a mechanism whereby the leader can relinquish it's lease. so i might write this:

> A new server is not allowed to serve as a leader (that is, serve up-to-date read requests or acknowledge write requests) until a lease of this duration has definitely expired on the old leader's side, or the old leader has explicitly acknowledged the new leader's lease.

though this also brushes over some details. so maybe a third option is:

> This lease allows the leader to safely serve reads for the duration of its lease, even during a network partition. See: https://docs.yugabyte.com/preview/architecture/transactions/single-row-transactions/#leader-leases-reading-the-latest-data-in-case-of-a-network-partition

**Review by robertsami — CHANGES_REQUESTED** [review]
why are we adding this flag in the yb-master docs and not yb-tserver?

**Comment by tylarb** [comment]
@robertsami  all of the other flags related here are actually in the master and tserver documentation - I believed that they should be the same in both places? Is that correct?

**Comment by robertsami** [comment]
> @robertsami all of the other flags related here are actually in the master and tserver documentation - I believed that they should be the same in both places? Is that correct?

leader_lease_duration_ms should be in both places with the same description

**Comment by robertsami** [comment]
> [](#commits-pushed-87af0f9)
> [ @ddhodge ](/ddhodge)
> ` [add to Tserver + stable 2.12 2.8](/yugabyte/yugabyte-db/pull/13800/commits/87af0f9c48468e13f883ad3190673b8adff5ba49) `
> ` [87af0f9](/yugabyte/yugabyte-db/pull/13800/commits/87af0f9c48468e13f883ad3190673b8adff5ba49) `
> [ @ddhodge ](/ddhodge)
> ` [Merge branch 'master' into docs_leader_lease_duration_flag](/yugabyte/yugabyte-db/pull/13800/commits/4818b602c540210694f3de516dbb559073cbbc10) `
> Verified
> ` [4818b60](/yugabyte/yugabyte-db/pull/13800/commits/4818b602c540210694f3de516dbb559073cbbc10) `
> [ @ddhodge ](/ddhodge)
> ` [format](/yugabyte/yugabyte-db/pull/13800/commits/7bb9c03a3d11d3866b93982bf80e7d7977f4ce9d) `
> ` [7bb9c03](/yugabyte/yugabyte-db/pull/13800/commits/7bb9c03a3d11d3866b93982bf80e7d7977f4ce9d) `
> [@robertsami](/robertsami)
> Member
> ### ** [robertsami](/robertsami)**

@tylarb are you asking if `leader_lease_duration_ms` must have the same value in both places? that i'm not sure of. i dont see why that would need to be the case, but @amitanandaiyer may be able to comment on that

**Comment by amitanandaiyer** [comment]
I don't think the raft/leader_lease values need to be the same across tservers and masters. They can be independently set.

However, I don't see a use case where they need to be different either.

**Comment by tylarb** [comment]
I did a quick review, and the changes LGTM at this point

**Pull request merged**